packages/components/src/components/menupopover/menupopover.component.ts

```diff
@@ -27,39 +30,168 @@ import styles from './menupopover.styles';
+      .filter((node) => !node.hasAttribute('disabled'));
+  }
+
   override connectedCallback() {
     super.connectedCallback();
     this.role = ROLE.MENU;
 
-    this.focusBackToTrigger = true;
     this.focusTrap = true;
     this.focusBackToTrigger = true;
     this.hideOnEscape = true;
     this.hideOnOutsideClick = true;
-    this.interactive = true;
     this.placement = POPOVER_PLACEMENT.BOTTOM_START;
     this.showArrow = false;
+    this.interactive = true;
     this.ariaOrientation = ORIENTATION.VERTICAL;
   }
 
   override async firstUpdated(changedProperties: PropertyValues) {
     await super.firstUpdated(changedProperties);
 
+    // Reset all tabindex to -1 and set the tabindex of the first menu item to 0
+    if (this.menuItems.length > 0) {
+      this.menuItems.forEach((menuitem) => menuitem.setAttribute('tabindex', '-1'));
+      this.menuItems[0].setAttribute('tabindex', '0');
+    }
+
     this.triggerElement?.setAttribute('aria-haspopup', ROLE.MENU);
     if (this.parentElement?.tagName?.toLowerCase() === MENU_POPOVER) {
-      this.interactive = true;
       this.placement = POPOVER_PLACEMENT.RIGHT_START;
     }
   }
 
+  private getCurrentIndex(target: EventTarget | null): number {
+    return this.menuItems.findIndex((node) => node === target);
+  }
+
+  private resetTabIndexAndSetFocus(newIndex: number, oldIndex: number) {
+    if (newIndex === oldIndex) return;
+    this.menuItems[oldIndex].setAttribute('tabindex', '-1');
+    this.menuItems[newIndex].setAttribute('tabindex', '0');
+    this.menuItems[newIndex].focus();
+  }
+
+  /**
+   * Closes all menu popovers in the stack.
+   * This method is used to ensure that when a menu item is clicked,
+   * all other open popovers are closed, maintaining a clean user interface.
+   * It iterates through the `popoverStack` and hides each popover until the stack is empty.
+   */
+  private closeAllMenuPopovers() {
+    while (popoverStack.peek()) {
+      const popover = popoverStack.pop();
+      if (popover) {
+        popover.hidePopover();
+      }
+    }
+  }
+
+  private handleMouseClick(event: MouseEvent): void {
+    const target = event.target as HTMLElement;
+    const role = target.getAttribute('role');
+    if (
+      isValidMenuItem(target)
+      && !isValidPopover(target.nextElementSibling)
+      && role === ROLE.MENUITEM
+    ) {
+      this.closeAllMenuPopovers();
+    }
+  }
+
+  private resolveDirectionKey(key: string, isRtl: boolean) {
+    if (!isRtl) return key;
+
+    switch (key) {
+      case KEYS.ARROW_LEFT:
+        return KEYS.ARROW_RIGHT;
+      case KEYS.ARROW_RIGHT:
+        return KEYS.ARROW_LEFT;
+      default:
+        return key;
+    }
+  }
+
+  private handleKeyDown(event: KeyboardEvent) {
+    const currentIndex = this.getCurrentIndex(event.target);
+    if (currentIndex === -1) return;
+
+    const isRtl = document.querySelector('html')?.getAttribute('dir') === 'rtl'
+     || window.getComputedStyle(this).direction === 'rtl';
```

Comment:
What is the point to test the HTML element as well? 
The consumer can change anywhere the direction of the text in the DOM. For example the HTML is RTL, but the menu (eg. for demo reason) is LTR.

This should be enough
```suggestion
    const isRtl = window.getComputedStyle(this).direction === 'rtl';
```